### PR TITLE
TFM integration with PSA tests

### PR DESCRIPTION
**Pull request by oyvindronningstad**
Just a first draft

Try any of the following, they should all run ok:

    west build samples/tfm_hello_world -p
    west build samples/tfm_regression_test -p
    west build samples/tfm_psa_test -p -- -DCONFIG_TFM_PSA_TEST_STORAGE=y

cc @ioannisg @magnev

### CODE REVIEW

File: samples/tfm_psa_test/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# SPDX-License-Identifier: Apache-2.0
+
+cmake_minimum_required(VERSION 3.13.1)
```
Comment (magnev) [inline samples/tfm_psa_test/CMakeLists.txt:3]: We should update the version requirement for CMake.
Some generator expressions did not work with CMake 3.13.3, but worked with CMake 3.18.4.

Comment (oyvindronningstad) [reply samples/tfm_psa_test/CMakeLists.txt:3]: Do you remember which generator expressions?

Comment (tejlmand) [reply samples/tfm_psa_test/CMakeLists.txt:3]: and we should only update to when those expressions were new, or even better, re-write them to `3.13.` versions if possible.

**Comment by oyvindronningstad** [comment]
Superseded by #3636

**Pull request closed without merging**